internal/checker/checker.go

```diff
@@ -41,89 +44,128 @@ func New(url, payload string, errors []string, client *resty.Client) Checker {
-			"security_level": "0",
-			"form":           "submit",
+			"login":    "admin",
+			"password": "12345",
+			"form":     "submit",
 		}).
 		Post("http://localhost/login.php")
 
 	if err != nil {
 		log.Printf("Could'n get cookies for bWAPP:%v\n", err)
 	}
 
-	//TODO: add kafka worker.
+	//TODO: add kafka worker. Here we'll get a URL.
 	return c
 }
 
 // Check firstly reads payloads from the file, then creates regexp for errors, after gets site body
-// checking it on possible inclusions of error keywords, finally posts form on site with injectable parameters.
+// checking it on possible inclusions of error keywords, finally posts htmlForm on site with injectable parameters.
 // Having this done matches response body with possible error keywords.
-func (c *checker) Check() {
+func (c *checker) Check(link string) error {
 	f, err := os.Open(c.payload)
 	if err != nil {
-		fmt.Println("Sorry could not parse the list -> ", c.payload)
-		return
+		return fmt.Errorf("sorry could not parse the list ->  %v\n", c.payload)
 	}
 	defer f.Close()
-	scan := bufio.NewScanner(f)
-	for scan.Scan() {
-		payloads := []string{
-			scan.Text(),
-		}
 
-		// send GET request to get body of a page before sending form.
-		// Page may have keywords of SQL errors.
-		resp, err := c.client.R().Get(c.url)
-		if err != nil {
-			log.Printf("Couldn't send get request: %v\n", err)
-			break
-		}
-		countBefore := c.countErrs(resp.Body())
-		var wg sync.WaitGroup
+	forms, countBefore, err := c.fetchForms(link)
+	fmt.Println("Forms received...")
+	if err != nil {
+		return err
+	}
+	var wg sync.WaitGroup
 
-		for _, payload := range payloads {
-			wg.Add(1)
-			go c.formSubmit(&wg, payload, countBefore)
-		}
-		wg.Wait()
+	scan := bufio.NewScanner(f)
+	for scan.Scan() {
+		payload := scan.Text()
+		wg.Add(1)
+		go c.submitForm(link, payload, countBefore, forms, setValues, &wg)
 	}
+	wg.Wait()
+	fmt.Println("Finished!")
+	return nil
 }
 
 // countErrs counts inclusions of errors from checker error-list if they are existed.
 func (c *checker) countErrs(bytes []byte) int {
-	var count = 0
+	var counter = 0
 	body := string(bytes)
 	for _, err := range c.errors {
-		count += strings.Count(body, err)
+		counter += strings.Count(body, err)
 	}
 
-	return count
+	return counter
 }
 
-// formSubmit submitting form putting each payload. Matches request body with possible errors.
-func (c *checker) formSubmit(wg *sync.WaitGroup, payload string, countBefore int) {
-	defer wg.Done()
-	resp, err := c.client.R().
-		SetFormData(map[string]string{
-			"login":    payload,
-			"password": "12345",
-			"form":     "submit",
-		}).
-		Post(c.url)
+//fetchForms fetches all forms which are exists in given link.
+func (c *checker) fetchForms(link string) ([]*htmlForm.HtmlForm, int, error) {
+	resp, err := c.client.R().Get(link)
+	if err != nil {
+		return nil, 0, fmt.Errorf("error fetching url %q: %v", link, err)
+	}
+
+	root, err := html.Parse(bytes.NewReader(resp.Body()))
```

Comment:
Pls, check if **resp.Body()** needs to be properly closed & **defer resp.Body().Close()** it if needed.